src/pages/iou/MoneyRequestSelectorPage.js

```diff
@@ -0,0 +1,139 @@
+};
+
+const defaultProps = {
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
+    iou: {
+        id: '',
+        amount: 0,
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
+    selectedTab: CONST.TAB.MANUAL,
+};
+
+function MoneyRequestSelectorPage(props) {
+    const iouType = lodashGet(props.route, 'params.iouType', '');
+    const reportID = lodashGet(props.route, 'params.reportID', '');
+    const {translate} = useLocalize();
+    const {canUseScanReceipts} = usePermissions();
+
+    const title = {
+        [CONST.IOU.MONEY_REQUEST_TYPE.REQUEST]: translate('iou.requestMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SEND]: translate('iou.sendMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SPLIT]: translate('iou.splitBill'),
+    };
+
+    const resetMoneyRequestInfo = () => {
+        const moneyRequestID = `${iouType}${reportID}`;
+        IOU.resetMoneyRequestInfo(moneyRequestID);
+    };
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            {({safeAreaPaddingBottomStyle}) => (
+                <FullPageNotFoundView shouldShow={!IOUUtils.isValidMoneyRequestType(iouType)}>
+                    <DragAndDropProvider isDisabled={props.selectedTab === CONST.TAB.MANUAL}>
+                        <View style={[styles.flex1, safeAreaPaddingBottomStyle]}>
+                            <HeaderWithBackButton
+                                title={title[iouType]}
+                                onBackButtonPress={Navigation.dismissModal}
+                            />
+                            {canUseScanReceipts ? (
+                                <OnyxTabNavigator
+                                    id={CONST.TAB.RECEIPT_TAB_ID}
+                                    tabBar={({state, navigation}) => (
+                                        <TabSelector
+                                            state={state}
+                                            navigation={navigation}
+                                            onTabPress={resetMoneyRequestInfo}
+                                        />
+                                    )}
+                                >
+                                    <TopTab.Screen
+                                        name={CONST.TAB.MANUAL}
+                                        component={MoneyRequestAmount}
+                                        initialParams={{reportID, iouType}}
+                                    />
+                                    <TopTab.Screen
+                                        name={CONST.TAB.SCAN}
+                                        component={ReceiptSelector}
+                                        initialParams={{reportID, iouType}}
+                                    />
+                                </OnyxTabNavigator>
+                            ) : (
+                                <MoneyRequestAmount />
```

Comment:
```suggestion
                                <MoneyRequestAmount route={props.route} />
```

This should fix the reported bug https://github.com/Expensify/App/pull/23046#issuecomment-1661220357. props.route was missing.

Reply:
Looking into this, the split button should not show the `MoneyRequestSelectorPage`, it should only show the `MoneyRequestAmount` component.

Reply:
Yes, I suggested an alternative fix without going into adding a new route